src/local/butler/reproduce.py

```diff
@@ -208,6 +209,28 @@ def _get_testcase_id_from_url(testcase_url):
   return testcase_id
 
 
+def _prepare_environment_for_android(testcase_path):
+  """Additional environment overrides needed to run on an Android device."""
+  environment.set_value('OS_OVERRIDE', 'ANDROID')
+
+  # Bail out if we don't have an Android device connected.
+  if not environment.get_value('ANDROID_SERIAL'):
+    # TODO(mbarbella): Handle the one-device case gracefully.
+    raise errors.ReproduceToolUnrecoverableError('Please set ANDROID_SERIAL.')
+
+  # TODO(mbarbella): Should we run device.initialize_device?
```

Comment:
This will be trivial to add, but some of the settings changes and other modifications seem invasive. Should we do it anyway?

Reply:
Maybe add a warning/prompt and then do this?

Reply:
+1 to prompt. And then we need to call initialize_device, otherwise things wont work, like setup_host_and_device_forwarder_if_needed, setup_asan_if_needed are all called from there. either can create another lightweight version of the function for just reproduce itself, like which calls e.g. setup_adb, etc.

Reply:
Added a warning. Based on Abhishek's point I think it's fine to always require that this happen to continue, but the prompt is still a great idea to make sure users know what they're getting into.